t/mojo/promise.t

```diff
@@ -160,15 +160,20 @@ subtest 'Double finally' => sub {
   is_deeply \@results, ['finally1', 'finally2'], 'promise not resolved';
 };
 
-subtest 'Resolved nested with finally' => sub {
-  my $promise  = Mojo::Promise->new;
-  my $promise2 = Mojo::Promise->new;
-  my @results;
-  $promise->finally(sub {$promise2})->finally(sub { @results = ('finally') });
-  $promise->resolve('pass');
-  Mojo::IOLoop->one_tick;
-  is_deeply \@results, ['finally'], 'promise already resolved';
-};
+# The expected behaviour here mismatches ES6: the 2nd finally() can’t
+# settle until the first one does, which in turn can’t settle until
+# $promise2 does. Since $promise2 doesn’t settle here, the 2nd finally()
+# should NOT fire.
+#
+#subtest 'Resolved nested with finally' => sub {
+#  my $promise  = Mojo::Promise->new;
+#  my $promise2 = Mojo::Promise->new;
+#  my @results;
+#  $promise->finally(sub {$promise2})->finally(sub { @results = ('finally') });
+#  $promise->resolve('pass');
+#  Mojo::IOLoop->one_tick;
+#  is_deeply \@results, ['finally'], 'promise already resolved';
+#};
```

Comment:
We never keep dead code around. That's a blocker. If you want to discuss this put it in the PR description (and perhaps explain it in the commit message).

Reply:
Also, when discussing specs it helps a lot to link to/quote the section you are talking about. JavaScript examples also simplify this significantly.